custom_components/ramses_cc/binary_sensor.py

```diff
@@ -1,34 +1,75 @@
 """Support for RAMSES binary sensors."""
 from __future__ import annotations
 
+from collections.abc import Mapping
+from dataclasses import dataclass
 from datetime import datetime as dt, timedelta
 import logging
+from types import UnionType
 from typing import Any
 
 from ramses_rf import Gateway
+from ramses_rf.device.base import BatteryState, HgiGateway
 from ramses_rf.device.heat import (
     SZ_CH_ACTIVE,
     SZ_CH_ENABLED,
     SZ_COOLING_ACTIVE,
     SZ_COOLING_ENABLED,
     SZ_DHW_ACTIVE,
+    SZ_DHW_BLOCKING,
     SZ_DHW_ENABLED,
     SZ_FAULT_PRESENT,
     SZ_FLAME_ACTIVE,
+    SZ_OTC_ACTIVE,
+    SZ_SUMMER_MODE,
+    Actuator,
+    TrvActuator,
 )
+from ramses_rf.entity_base import Entity as RamsesRFEntity
+from ramses_rf.schemas import SZ_SCHEMA
+from ramses_rf.system.heat import Logbook, System
 from ramses_tx.const import SZ_BYPASS_POSITION, SZ_IS_EVOFW3
 
 from homeassistant.components.binary_sensor import (
+    ENTITY_ID_FORMAT,
     BinarySensorDeviceClass,
     BinarySensorEntity,
+    BinarySensorEntityDescription,
 )
+from homeassistant.const import EntityCategory
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import RamsesSensorBase
+from . import RamsesEntity, RamsesEntityDescription
 from .broker import RamsesBroker
-from .const import ATTR_BATTERY_LEVEL, BROKER, DOMAIN
+from .const import (
+    ATTR_ACTIVE_FAULT,
+    ATTR_BATTERY_LEVEL,
+    ATTR_LATEST_EVENT,
+    ATTR_LATEST_FAULT,
+    ATTR_SCHEMA,
+    BROKER,
+    DOMAIN,
+)
+
+
+@dataclass(kw_only=True)
+class RamsesBinarySensorEntityDescription(
+    RamsesEntityDescription, BinarySensorEntityDescription
+):
+    """Class describing Ramses binary sensor entities."""
+
+    attr: str | None = None
```

Comment:
This isn't currently used but I added it to allow the description key and device attribute to differ.

This would then allow for a HA entity id of `..._temperature` (taken from key) and a ramses_rf device attribute of `..._temp` (taken from attr). I see this has happened in the past but as far as I could tell is not currently the case anywhere.

Reply:
Yes - was a pain, so I renamed them all. 

Regardless, there's a good chance we'll see something similar in the future.